presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -164,8 +164,44 @@ public PlanNode visitPlan(PlanNode node, RewriteContext<Expression> context)
         @Override
         public PlanNode visitProject(ProjectNode node, RewriteContext<Expression> context)
```

Comment:
Add a comment explaining what the high level strategy is. E.g., "push down conjuncts from the inherited predicate that don't depend on non-deterministic assignments"